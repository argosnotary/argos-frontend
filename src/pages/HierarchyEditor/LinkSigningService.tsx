import {signString} from "../../security";
import {ILink, ILinkMetaBlock} from "../../interfaces/ILink";
import stringify from "json-stable-stringify";

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const signLink = async (
    password: string,
    keyId: string,
    encryptedPrivateKey: string,
    link: ILink,
    supplyChainId: string
): Promise<ILinkMetaBlock> => {
    const signature = await signString(
        password,
        encryptedPrivateKey,
        serialize(link)
    );
    return {
        supplyChainId: supplyChainId,
        signature: { signature: signature, keyId: keyId },
        link: link
    };
};

const serialize = (link: ILink): string => {
    return stringify(link);
};

export { serialize,  signLink};