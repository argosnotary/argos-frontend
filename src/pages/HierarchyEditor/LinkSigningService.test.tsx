/*
 * Copyright (C) 2020 Argos Notary
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from "react";

import {serialize} from "./LinkSigningService";

it("test serialize", () => {
    const seralized = serialize({
        runId: "69af2dd",
        layoutSegmentName: "segment1",
        materials: [
            {
                uri: "src/test/java/com/argosnotary/argos/test/ArgosServiceTestIT.java",
                hash: "61a0af2b177f02a14bab478e68d4907cda4dc3f642ade0432da8350ca199302b"
            }
        ],
        stepName: "build",
        products: [
            {
                uri: "src/test/java/com/argosnotary/argos/test/ArgosServiceTestIT.java",
                hash: "61a0af2b177f02a14bab478e68d4907cda4dc3f642ade0432da8350ca199302b"
            }
        ]
    });

    expect(seralized).toMatchSnapshot();
});
