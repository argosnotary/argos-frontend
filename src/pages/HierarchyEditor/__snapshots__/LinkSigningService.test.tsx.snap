// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test serialize 1`] = `"{\\"layoutSegmentName\\":\\"segment1\\",\\"materials\\":[{\\"hash\\":\\"61a0af2b177f02a14bab478e68d4907cda4dc3f642ade0432da8350ca199302b\\",\\"uri\\":\\"src/test/java/com/argosnotary/argos/test/ArgosServiceTestIT.java\\"}],\\"products\\":[{\\"hash\\":\\"61a0af2b177f02a14bab478e68d4907cda4dc3f642ade0432da8350ca199302b\\",\\"uri\\":\\"src/test/java/com/argosnotary/argos/test/ArgosServiceTestIT.java\\"}],\\"runId\\":\\"69af2dd\\",\\"stepName\\":\\"build\\"}"`;
